### ISSUE TITLE
Problem with training on Waymo

### ISSUE THREAD

**Issue by mive93** [maudzung/SFA3D#15]
Dear @maudzung,

first, thank you for your work.
@lucabart97 and I, have spent the last week trying to train this network on the Waymo dataset. 
We first used [your other repo](https://github.com/maudzung/convert_waymo_2_kitti) to convert Waymo to Kitti format.
Then we changed the train.txt and val.txt files and also [kitti_config.txt](https://github.com/maudzung/SFA3D/files/5428196/kitti_config.txt).
However, we were not able to achieve any good results with the training. 
Indeed, the train loss has a very weird behavior:

![trainloss](https://user-images.githubusercontent.com/11939259/96972652-0eb36100-1517-11eb-8542-9da23a9751df.png)

Can you give us some tips to make it work? Or point out what we are doing wrong?

Thank you very much in advance. 

Best,
Micaela

**Comment by maudzung** [comment]
Hi @mive93 
Did you carefully visualize and check the converted dataset?

**Comment by mive93** [comment]
Yes, we did. 
We visualized everything (cloud, images and labels) and the conversion was okay.

**Comment by maudzung** [comment]
Because of my limited resource, I didn't try to train the network on the Waymo dataset. I thought I should be trained well on the Waymo dataset also. There are 5 components in the total loss, you can check each curve to check which components aren't changed over time.

**Comment by mive93** [comment]
Thank you for your answer. 
I have checked the components, and the one causing problem is train_hm_cen_loss, that always has a fixed value.
Do you think it's related to the conversion?

**Comment by MukundVarmaT** [comment]
Hi! I have noticed a similar issue while training on the Kitti Dataset but not always. During some runs, the loss starts reducing but after some epochs the center heatmap loss increases and stays constant while in some other cases it keeps increasing and decreasing. I have also noticed a similar case while changing resnet-18 to resnet-34 and I am not sure why. 

Some failure cases:
![Screenshot from 2020-10-28 13-32-57](https://user-images.githubusercontent.com/47249476/97408547-3ee96e00-1922-11eb-9f96-01d4b21f840f.png)
![Screenshot from 2020-10-28 13-32-38](https://user-images.githubusercontent.com/47249476/97408548-414bc800-1922-11eb-96ec-15717df1887f.png)

**Comment by maudzung** [comment]
Let me check again about the center heatmap generation to see what happended.

**Comment by zmin666666** [comment]
> Thank you for your answer.
> I have checked the components, and the one causing problem is train_hm_cen_loss, that always has a fixed value.
> Do you think it's related to the conversion?

Hi, @mive93, @maudzung 

I also tried to train the SFA3D in the Waymo dataset and meet the same loss problem when using resnet34 and resnet50. If I use loss = loss - (pos_loss/num_pos + neg_loss/num_neg) and set the LR rate to 0.0005, the loss problem will be solved but the prediction res will be worse. Have you solved this problem?

Zhimin Chen

**Comment by maudzung** [comment]
I think the problem is related to the reported issue [here](https://github.com/maudzung/RTM3D/issues/10). I'll try to solve the problem in the near future.

**Comment by SofianeB-03** [comment]
Hello @Zhimin-C , Can you tell me what is the values of the boundary min_Z and max_Z that you have used for Waymo please ? Because I think the range of Z-axis has an impact in the predictions according the dataset (the height of the lidar).

Thanks.

**Comment by Zhimin-C** [comment]
Hi,

https://arxiv.org/pdf/2006.15505.pdf
This HorizonLiDAR3D is the top method on the Waymo dataset 3D detection
challenge:
https://waymo.com/open/challenges/2020/3d-detection/
You can refer to their parameters in the experiment part. The height range
is -1 to 3.

Zhimin Chen

**Comment by SofianeB-03** [comment]
@Zhimin-C Thank you very much for your quick response !
